static/src/js/actions/granules.js

```diff
@@ -223,51 +226,66 @@ export const fetchLinks = retrievalCollectionData => (dispatch, getState) => {
         temporal: $temporal
         twoDCoordinateSystem: $twoDCoordinateSystem
       ) {
-        items{
+        cursor
+        items {
           links
         }
       }
     }`
 
-  return Promise.all(Array.from(Array(totalPages)).map((_, pageNum) => {
-    const granuleResponse = graphQlRequestObject.search(graphQuery, {
-      ...preparedGranuleParams,
-      limit: pageSize,
-      offset: pageSize * pageNum,
-      linkTypes: ['data', 's3'],
-      collectionConceptId: collectionId
-    })
-      .then((response) => {
-        const { data } = response
-        const { data: granulesData } = data
-        const { granules } = granulesData
-        const { items } = granules
-
-        if (items) {
-          // Fetch the download links from the granule metadata
-          const granuleDownloadLinks = getDownloadUrls(items)
-          const granuleS3Links = getS3Urls(items)
-
-          dispatch(updateGranuleLinks({
-            id,
-            links: {
-              download: granuleDownloadLinks.map(lnk => lnk.href),
-              s3: granuleS3Links.map(lnk => lnk.href)
-            }
-          }))
-        }
+  let cursor
+  let response
+
+  try {
+    // Request an additional page (totalPages + 1) when scrolling to force GraphQL to clear the scroll session
+    await Array.from(Array(totalPages + 1)).forEachAsync(async (_, currentPage) => {
```

Comment:
I'm hesitant to implement scroll sessions using known page numbers over looping until we get nothing because thats how CMR expects it to work -- technically speaking the results could change while paging, I guess? I think we should change this to looping until its empty, even though the progress bar uses the page numbers for its math, I don't see a way around that.